core/src/ibc/transaction_handler/mod.rs

```diff
@@ -132,6 +133,113 @@ pub fn execute(
                 .handle_open_confirm(identifier, proof_ack, proof_height)
                 .map_err(|err| RuntimeError::IBC(format!("ConnOpenConfirm: {}", err)).into())
         }
-        _ => unimplemented!(),
+        Datagram::ChanOpenInit {
+            raw,
+        } => {
+            let mut channel_manager = ibc_channel::Manager::new(&mut context);
+            channel_manager
+                .chan_open_init(
+                    {
+                        if raw.order == 1 {
+                            ibc::channel_04::types::ChannelOrder::ORDERED
+                        } else {
+                            ibc::channel_04::types::ChannelOrder::UNORDERED
+                        }
+                    },
+                    raw.connection,
+                    raw.channel_identifier,
+                    raw.counterparty_channel_identifier,
+                    raw.version,
+                )
+                .map_err(|err| RuntimeError::IBC(format!("ChanOpenInit: {}", err)).into())
+                .map(|_| ())
```

Comment:
Why did you add ` .map(|_| ())` here?
How about this?

```
            channel_manager
                .chan_open_init(
                    {
                        if raw.order == 1 {
                            ibc::channel_04::types::ChannelOrder::ORDERED
                        } else {
                            ibc::channel_04::types::ChannelOrder::UNORDERED
                        }
                    },
                    raw.connection,
                    raw.channel_identifier,
                    raw.counterparty_channel_identifier,
                    raw.version,
                )
                .map_err(|err| RuntimeError::IBC(format!("ChanOpenInit: {}", err)).into())?;
            Ok(())
```

Reply:
That explicitly discards any returned result. I have no opinion about that, so I'll just follow your suggestion.